src/lib/isTaxID.js

```diff
@@ -54,20 +442,815 @@ function enUsCheck(tin) {
+}
+
+/*
+ * lt-LT validation function (should go here if needed)
+ * (Asmens kodas, persons/entities respectively)
+ * Current validation check is alias of etEeCheck- same format applies
+ */
+
+/*
+ * it-IT first/last name validity check
+ * Accepts it-IT TIN-encoded names as a three-element character array and checks their validity
+ * Due to lack of clarity between resources ("Are only Italian consonants used?
+ * What happens if a person has X in their name?" etc.) only two test conditions
+ * have been implemented:
+ * Vowels may only be followed by other vowels or an X character
+ * and X characters after vowels may only be followed by other X characters.
+ */
+function itItNameCheck(name) {
+  // true at the first occurence of a vowel
+  let vowelflag = false;
+
+  // true at the first occurence of an X AFTER vowel
+  // (to properly handle last names with X as consonant)
+  let xflag = false;
+
+  for (let i = 0; i < 3; i++) {
+    if (!vowelflag && /[AEIOU]/.test(name[i])) {
+      vowelflag = true;
+    } else if (!xflag && vowelflag && (name[i] === 'X')) {
+      xflag = true;
+    } else if (i > 0) {
+      if (vowelflag && !xflag) {
+        if (!/[AEIOU]/.test(name[i])) { return false; }
+      }
+      if (xflag) {
+        if (!/X/.test(name[i])) { return false; }
+      }
+    }
+  }
+  return true;
+}
+
+/*
+ * it-IT validation function
+ * (Codice fiscale (TIN-IT), persons only)
+ * Verify name, birth date and codice catastale validity
+ * and calculate check character.
+ * Material not in DG-TAXUD document sourced from:
+ * `https://en.wikipedia.org/wiki/Italian_fiscal_code`
+ */
+function itItCheck(tin) {
+  // Capitalize and split characters into an array for further processing
+  const chars = tin.toUpperCase().split('');
+
+  // Check first and last name validity calling itItNameCheck()
+  if (!itItNameCheck(chars.slice(0, 3))) { return false; }
+  if (!itItNameCheck(chars.slice(3, 6))) { return false; }
+
+  // Convert letters in number spaces back to numbers if any
+  const number_locations = [6, 7, 9, 10, 12, 13, 14];
+  const number_replace = {
+    L: '0',
+    M: '1',
+    N: '2',
+    P: '3',
+    Q: '4',
+    R: '5',
+    S: '6',
+    T: '7',
+    U: '8',
+    V: '9',
+  };
+  for (const i of number_locations) {
+    if (chars[i] in number_replace) {
+      chars.splice(i, 1, number_replace[chars[i]]);
+    }
+  }
+
+  // Extract month and day, and check date validity
+  const month_replace = {
+    A: '01',
+    B: '02',
+    C: '03',
+    D: '04',
+    E: '05',
+    H: '06',
+    L: '07',
+    M: '08',
+    P: '09',
+    R: '10',
+    S: '11',
+    T: '12',
+  };
+  let month = month_replace[chars[8]];
+
+  let day = parseInt(chars[9] + chars[10], 10);
+  if (day > 40) { day -= 40; }
+  if (day < 10) { day = `0${day}`; }
+
+  const date = `${chars[6]}${chars[7]}/${month}/${day}`;
+  if (!isDate(date, 'YY/MM/DD')) { return false; }
+
+  /*
+   * Check validity of birth town registry number (codice catastale)
```

Comment:
May be we separate this section as a different PR so that we give it more deliberation; but then allow the rest to land?